wallet/src/account/output_cache/mod.rs

```diff
@@ -680,56 +684,86 @@ impl OutputCache {
         ))
     }
 
-    pub fn check_conflicting(&mut self, tx: &WalletTx, block_id: Id<GenBlock>) -> Vec<&WalletTx> {
-        let is_unconfirmed = match tx.state() {
-            TxState::Inactive(_)
-            | TxState::InMempool(_)
-            | TxState::Conflicted(_)
-            | TxState::Abandoned => true,
-            TxState::Confirmed(_, _, _) => false,
-        };
+    pub fn update_conflicting_txs(
+        &mut self,
+        confirmed_tx: &Transaction,
+        block_id: Id<GenBlock>,
+    ) -> WalletResult<Vec<Id<Transaction>>> {
+        let mut frozen_token_id: Option<TokenId> = None;
+        let mut confirmed_account_nonce: Option<(AccountType, AccountNonce)> = None;
 
-        if is_unconfirmed {
-            return vec![];
+        for input in confirmed_tx.inputs() {
+            match input {
+                TxInput::Utxo(_) => {
+                    //TODO: check conflicting utxo spends
+                }
+                TxInput::Account(outpoint) => {
+                    confirmed_account_nonce = Some((outpoint.account().into(), outpoint.nonce()));
+                }
+                TxInput::AccountCommand(nonce, cmd) => {
+                    confirmed_account_nonce = Some((cmd.into(), *nonce));
+                    match cmd {
+                        AccountCommand::MintTokens(_, _)
+                        | AccountCommand::UnmintTokens(_)
+                        | AccountCommand::LockTokenSupply(_)
+                        | AccountCommand::ChangeTokenMetadataUri(_, _)
+                        | AccountCommand::ChangeTokenAuthority(_, _)
+                        | AccountCommand::UnfreezeToken(_)
+                        | AccountCommand::ConcludeOrder(_)
+                        | AccountCommand::FillOrder(_, _, _) => { /* do nothing */ }
+                        | AccountCommand::FreezeToken(token_id, _) => {
+                            frozen_token_id = Some(*token_id);
+                        }
+                    }
+                }
+            }
```

Comment:
If only one Account or AccountCommand can be present in the tx, maybe we can replace it with a find() that returns the 2 Options.